src/bitcoin/utilities/accounts/HDSegwitWallet.ts

```diff
@@ -606,9 +606,13 @@ export default class HDSegwitWallet extends Bitcoin {
         this.derivativeAccounts[ dAccountType ][ accountNumber ][
           'usedAddresses'
         ] = usedAddresses // derv used addresses forms a part of ownedAddresses array during primary-acc sync
-        // console.log({ derivativeAccUsedAddresses: usedAddresses });
-
-        const { xpubId } = this.derivativeAccounts[ dAccountType ][ accountNumber ]
+        
+        const xpub = this.generateDerivativeXpub( dAccountType, accountNumber )
```

Comment:
Looks good, only thing is that we don't need to regenerate the xpub(would end up generating a different xpub), xpub can be pulled out of the derivative acccount using the following:
```
const { xpub, xpubId } = this.derivativeAccounts[ dAccountType ][ accountNumber]
if ( !xpubId ) {
          xpubId = crypto.createHash( 'sha256' ).update( xpub ).digest( 'hex' )
          this.derivativeAccounts[ dAccountType ][ accountNumber ].xpubId = xpubId
        }
```

Reply:
Kool, updated in new commit.